hydra-tx/src/Hydra/Tx/HeadId.hs

```diff
@@ -1,18 +1,16 @@
 {-# LANGUAGE DerivingVia #-}
 
-module Hydra.HeadId where
+module Hydra.Tx.HeadId where
```

Comment:
Should: move into `hydra-tx`, but keep it separate to "main-chain specific" stuff what `Hydra.Tx` actually is.

Concretely, just keep this to be `Hydra.HeadId`.

Why? We will move it further "upstream", e.g. into a `hydra-api` (`https://github.com/cardano-scaling/hydra/discussions/432#discussioncomment-3219462`) or even a `hydra-base` types package.

Reply:
This applies to `Party`, `ContestatioPeriod`, etc. too

Reply:
I don't agree with this. `hydra-base` is not good design to me. The types are only significant in so far as they are used in the transaction construction. I would like to keep it namespaced under `Hydra.Tx`, and `Hydra.API` can re-export for convenience.

Reply:
I don't agree with this. `hydra-base` is not good design to me. The types are only significant in so far as they are used in the transaction construction. I would like to keep it namespaced under `Hydra.Tx`, and `Hydra.API` can re-export for convenience.

Reply:
Do you agree that these types are non-cardano-specific at least?

To me `Hydra.Tx` is a "cardano-specific implementation of the Hydra protocol transactions (L1)". So that is why I would like to keep `Hydra.HeadId` (and others moved) separate to that namespace. The `HeadId` (and other types) are not cardano-specific, but special to the Hydra (head) protocol design. Another good example is `ContestationPeriod`, it has nothing to do with transactions nor Cardano.

Reply:
I don't disagree. I don't think anything is really gained by separating that out at the moment though.